### ISSUE TITLE
AutoSizer does not show list items with scrollToIndex

### ISSUE THREAD

**Issue by ethankong113** [bvaughn/react-virtualized#691]
Demo here: https://plnkr.co/edit/6GFoRcQrM3jdy3aYGvly?p=preview

I have a list that pans out when a user clicks the open button. However, when I use it with scrollToIndex to scroll to a specific list item, it seems to hide the items that are above the index. And if you change the index to a larger number, the entire list disappears. This problem goes away when you scroll onto the list, but it returns you to the top instead of the indexed item.

**Comment by bvaughn** [comment]
Not sure what the demo is showing but it's not using `AutoSizer` correctly. (The `width` and `height` parameters are ignored in favor of hard-coded values.) Here's a fork that seems to work fine: https://plnkr.co/edit/5NsaqR?p=preview

Closing this for now since it seems to work okay for me. We can keep talking here though if I'm overlooking something. 😄

**Comment by ethankong113** [comment]
Are you saying that the AutoSizer can only pan out vertically instead of horizontally?

Did you see how the demo doesn't show the menu items before the scrollToIndex? Or you can check this one (https://plnkr.co/edit/uYtFXdNNw1a5CRwKgaO8?p=preview). When I pick a large scrollToIndex number... the entire list is missing.

**Comment by bvaughn** [comment]
No. I was juust saying that the original Plnkr is strange. This is not how `AutoSizer` should be used:
```jsx
<AutoSizer>
  {({height, width}) => ( // height and width are not used, so why use AutoSizer?
    <List
      width={this.state.open ? 500 : 0}
      height={700}
```

I think the problem you're reporting is that the combination of setting `scrollToIndex` with a `height` and/or `width` of 0 causes the `scrollToIndex` property to be ignored, since no cells are rendered when `height` or `width` are 0.

If you change...
```js
width={this.state.open ? 500 : 0}
```
...to...
```js
width={this.state.open ? 500 : 1}
```
...your example works. I'll re-open. Feel free to submit a PR if you'd like. 😄

**Comment by ethankong113** [comment]
Yes. I tried to do 1 instead of 0 and used a CSS visibility to hide the list. It works but it seems to be a hack.

**Comment by bvaughn** [comment]
Yeah. I was just thrown by the original Plnkr b'c it seemed like it was doing things that weren't valid.

**Comment by ethankong113** [comment]
I will see if there's a fix for this over the weekend :)

**Comment by bvaughn** [comment]
Resolved via 67f9dc5

**Comment by bvaughn** [comment]
Fix released with 9.8.0
